### PR TITLE
Close regression data files 

### PR DESCRIPTION
**Pull request by atharva-2001**
### :pencil: Description

**Type:** :beetle: `bugfix` | :vertical_traffic_light: `testing` 

Closes regression data files after tests have run and suppresses warnings.

### :pushpin: Resources

Examples, notebooks, and links to useful references.

### :vertical_traffic_light: Testing

How did you test these changes?

- [X] Testing pipeline
- [ ] Other method (describe)
- [ ] My changes can't be tested (explain why)

### :ballot_box_with_check: Checklist

- [ ] I requested two reviewers for this pull request
- [ ] I updated the documentation according to my changes
- [ ] I built the documentation by applying the `build_docs` label

> **Note:** If you are not allowed to perform any of these actions, ping (@) a contributor.

### CODE REVIEW

**Review by KasukabeDefenceForce — APPROVED** [review]

**Review by andrewfullard — APPROVED** [review]

**Comment by andrewfullard** [comment]
Looks like this isn't working as intended right now

File: tardis/tests/fixtures/regression_data.py [hunk 1]
```diff
@@ -161,9 +162,14 @@ def sync_hdf_store(self, tardis_module, update_fname=True):
                 f"Skipping test to generate regression data: {self.fpath}"
             )
         else:
-            return pd.HDFStore(self.fpath, mode="r")
-
+            # since each test function has its own regression data instance
+            # each test function will only have one HDFStore object
+            self.hdf_store_object = pd.HDFStore(self.fpath, mode="r")
+            return self.hdf_store_object
```
Comment (KasukabeDefenceForce) [inline tardis/tests/fixtures/regression_data.py:168]: Should we just create a dictionary here and update the tests accordingly?

**Review by andrewfullard — APPROVED** [review]

**Comment by andrewfullard** [comment]
Still a lot of unclosed file warnings 🤔

File: tardis/tests/fixtures/regression_data.py (hunk 1 above)
Comment (atharva-2001) [reply tardis/tests/fixtures/regression_data.py:168]: Since there is only one file for each test I don't think dict would do much help right?

**Review by andrewfullard — APPROVED** [review]

**Review by DeerWhale — APPROVED** [review]

**Pull request merged**